In [4]:
import pandas as pd
import numpy as np
import json
import requests

Question 1
What is the result of the following code?

import pandas as pd
import numpy as np

df = pd.DataFrame(np.random.randn(5, 3), columns=['col1', 'col2', 'col3'])
df.apply(lambda x: x.max() - x.min())
A. col1 1.0 col2 1.0 col3 1.0 dtype: float64

B. col1 1.0 col2 1.0 col3 1.0 dtype: float64

C. col1 1.0 col2 1.0 col3 1.0 dtype: float64

D. col1 1.0 col2 1.0 col3 1.0 dtype: float64

In [2]:
df = pd.DataFrame(np.random.randn(5, 3), columns=['col1', 'col2', 'col3'])
df.apply(lambda x: x.max() - x.min())

col1    2.231562
col2    2.760107
col3    2.118669
dtype: float64

Question 2
Download a csv file from https://stats.govt.nz/assets/Uploads/Business-employment-data/Business-employment-data-June-2022-quarter/Download-data/business-employment-data-june-2022-quarter-csv.zip and load it into a pandas dataframe. How many rows and columns are there?

In [3]:
df = pd.read_csv(r'C:\Users\new\Desktop\Week12\machine-readable-business-employment-data-june-2022-quarter.csv')
df

Series_reference   Period  Data_value Suppressed STATUS   UNITS  \
0          BDCQ.SEA1AA  2011.06     80078.0        NaN      F  Number   
1          BDCQ.SEA1AA  2011.09     78324.0        NaN      F  Number   
2          BDCQ.SEA1AA  2011.12     85850.0        NaN      F  Number   
3          BDCQ.SEA1AA  2012.03     90743.0        NaN      F  Number   
4          BDCQ.SEA1AA  2012.06     81780.0        NaN      F  Number   
...                ...      ...         ...        ...    ...     ...   
20249    BDCQ.SEE3999A  2017.06         NaN          Y      C  Number   
20250    BDCQ.SEE3999A  2017.09         NaN          Y      C  Number   
20251    BDCQ.SEE3999A  2017.12         NaN          Y      C  Number   
20252    BDCQ.SEE3999A  2018.03         NaN          Y      C  Number   
20253    BDCQ.SEE3999A  2018.06         NaN          Y      C  Number   

       Magnitude                         Subject  \
0              0  Business Data Collection - BDC   
1              0  Business Data Collection - BDC   
2              0  Business Data Collection - BDC   
3              0  Business Data Collection - BDC   
4              0  Business Data Collection - BDC   
...          ...                             ...   
20249          0  Business Data Collection - BDC   
20250          0  Business Data Collection - BDC   
20251          0  Business Data Collection - BDC   
20252          0  Business Data Collection - BDC   
20253          0  Business Data Collection - BDC   

                                              Group  \
0                   Industry by employment variable   
1                   Industry by employment variable   
2                   Industry by employment variable   
3                   Industry by employment variable   
4                   Industry by employment variable   
...                                             ...   
20249  Territorial authority by employment variable   
20250  Territorial authority by employment variable   
20251  Territorial authority by employment variable   
20252  Territorial authority by employment variable   
20253  Territorial authority by employment variable   

                               Series_title_1  \
0                                 Filled jobs   
1                                 Filled jobs   
2                                 Filled jobs   
3                                 Filled jobs   
4                                 Filled jobs   
...                                       ...   
20249  Filled jobs (workplace location based)   
20250  Filled jobs (workplace location based)   
20251  Filled jobs (workplace location based)   
20252  Filled jobs (workplace location based)   
20253  Filled jobs (workplace location based)   

                           Series_title_2 Series_title_3  Series_title_4  \
0       Agriculture, Forestry and Fishing         Actual             NaN   
1       Agriculture, Forestry and Fishing         Actual             NaN   
2       Agriculture, Forestry and Fishing         Actual             NaN   
3       Agriculture, Forestry and Fishing         Actual             NaN   
4       Agriculture, Forestry and Fishing         Actual             NaN   
...                                   ...            ...             ...   
20249  Area Outside Territorial Authority         Actual             NaN   
20250  Area Outside Territorial Authority         Actual             NaN   
20251  Area Outside Territorial Authority         Actual             NaN   
20252  Area Outside Territorial Authority         Actual             NaN   
20253  Area Outside Territorial Authority         Actual             NaN   

       Series_title_5  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4                 NaN  
...               ...  
20249             NaN  
20250             NaN  
20251             NaN  
20252             NaN  
20253             NaN  

[20254 rows x 14 columns]

20254 rows

14 columns

Question 3
What method can be used to get the number of non-NA values in a pandas dataframe?

A. df.count()

B. df.na_count()

C. df.na_values()

D. df.na_count()

Answer: A 

Question 4
Create a dataframe with 10 rows and 3 columns, where the values are random numbers between 1 and 10 (inclusive). Then, replace all values greater than 5 with the value 10.

In [28]:
random_df = pd.DataFrame(np.random.randint(0,11,size=(10, 3)), columns=list('XYZ'))
random_df


X  Y   Z
0   8  5  10
1   7  1   9
2   4  8   9
3   0  4   2
4   6  5   5
5   8  1   8
6   7  6   6
7  10  3   3
8   0  3   1
9   5  9   3

In [29]:
#REPLACE
random_df.loc[random_df['X'] > 5] = 10
random_df.loc[random_df['Y'] > 5] = 10
random_df.loc[random_df['Z'] > 5] = 10
random_df


X   Y   Z
0  10  10  10
1  10  10  10
2  10  10  10
3   0   4   2
4  10  10  10
5  10  10  10
6  10  10  10
7  10  10  10
8   0   3   1
9  10  10  10

Question 5
create a dataframe from this link https://jsonplaceholder.typicode.com/albums

In [9]:
url = 'https://jsonplaceholder.typicode.com/albums'
df = pd.read_json(url)
df



userId   id                                              title
0        1    1                              quidem molestiae enim
1        1    2                   sunt qui excepturi placeat culpa
2        1    3                                 omnis laborum odio
3        1    4           non esse culpa molestiae omnis sed optio
4        1    5                                  eaque aut omnis a
..     ...  ...                                                ...
95      10   96  unde et ut molestiae est molestias voluptatem ...
96      10   97                                       est quod aut
97      10   98  omnis quia possimus nesciunt deleniti assumend...
98      10   99    consectetur ut id impedit dolores sit ad ex aut
99      10  100                                 enim repellat iste

[100 rows x 3 columns]